lucene/core/src/test/org/apache/lucene/util/graph/TestGraphTokenStreamFiniteStrings.java

```diff
@@ -660,4 +663,27 @@ public void testMultipleSidePathsWithGaps() throws Exception {
         it.next(), new String[] {"king", "alfred", "saxons", "ruled"}, new int[] {1, 1, 3, 1});
     assertFalse(it.hasNext());
   }
+
+  public void testLongTokenStreamStackOverflowError() throws Exception {
+
+    ArrayList<Token> tokens =
+        new ArrayList<Token>() {
+          {
+            add(token("fast", 1, 1));
+            add(token("wi", 1, 1));
+            add(token("wifi", 0, 2));
+            add(token("fi", 1, 1));
+          }
+        };
+
+    // Add in too many tokens to get a high depth graph
+    for (int i = 0; i < 1024 * 10; i++) {
+      tokens.add(token("network", 1, 1));
+    }
+
+    TokenStream ts = new CannedTokenStream(tokens.toArray(new Token[0]));
+    GraphTokenStreamFiniteStrings graph = new GraphTokenStreamFiniteStrings(ts);
+
+    assertThrows(IllegalArgumentException.class, () -> graph.articulationPoints());
```

Comment:
nit: we prefer method refs to lambdas whenever possible

```suggestion
    assertThrows(IllegalArgumentException.class, graph::articulationPoints);
```